[FEAT] Adds auto filename context on new session chats where the chat box is empty

charlwillia6 opened the pull request:
## Description ✏️

Please review https://github.com/trypear/pearai-submodule/pull/91
That PR was merged and then reverted due to an issue where all text and context would be removed from the chat input box if there was any error.  This PR still has the same functionality as the previous PR, but fixes this issue.  

It also fixes the issue with slash commands getting overwritten when the auto context feature is triggered.  

You can review comments and screenshots in PR 91.

This feature is ready for testing.

## Checklist ✅

- [ ] I have added screenshots (if UI changes are present).
- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
[FEAT] Add active editor tracking and context item updates based on file changes

• Auto-add file mentions on editor change
• Improve context item cleanup on send
• Add filepath validation utilities
• Debounce editor change notifications
• Fix mention handling in TipTap editor

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
[FIX] Updated the isEmpty function to check for context length so that context does not get overwritten.

charlwillia6 committed:
[FIX] getSuggestions was causing flushSync errors, refactored to fix those errors

charlwillia6 committed:
[FIX] Issues with auto context when clicking return in the chat input box

• Move context items to component level
• Clean up mention node handling
• Improve clipboard event handling
• Optimize enter key handler
• Update context item persistence

charlwillia6 committed:
Clean-up

Fryingpannn committed:
Fix empty editory check - add slashcommand and codeblocks

Fryingpannn committed:
Revert some refactoring

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
Merge branch 'charlwillia6/add-file-to-context-when-no-text-selected' of https://github.com/charlwillia6/pearai-submodule into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'origin/main' into charlwillia6/add-file-to-context-when-no-text-selected

Himanshu-Singh-Chauhan committed:
fix: clear main input after response completes

Himanshu-Singh-Chauhan committed:
undo refactoring

Himanshu-Singh-Chauhan committed:
reverts

nang-dev2 committed:
Merge branch 'main' of https://github.com/trypear/pearai-submodule into charlwillia6/add-file-to-context-when-no-text-selected

nang-dev2 committed:
Merge branch 'charlwillia6/add-file-to-context-when-no-text-selected' of https://github.com/charlwillia6/pearai-submodule into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
Move & improve file path handling in code editor for UX enhancements

• Centralized isValidFilePath utility
• Added newSession command handler
• Streamlined editor change handling
• Fixed focus management in editor

Himanshu-Singh-Chauhan committed:
move newsession command to commands.ts file

Himanshu-Singh-Chauhan committed:
add currentfile on onDidChangeActiveTextEditor

Commit:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge branch 'main' into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/add-file-to-context-when-no-text-selected

Fryingpannn committed:
Remove debounce

Fryingpannn committed:
Merge branch 'main' into charlwillia6/add-file-to-context-when-no-text-selected

Fryingpannn committed:
Add back memo

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge remote-tracking branch 'origin/main' into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
[FIX] Chat input editor interaction and context handling

• Added error handling for chat input with existing text
• Fixed bug where clicking editor would not add file as context
• Improved input validation for chat context
• Better handling of editor state changes
• Fixed issue with slash commands getting overwritten

Fryingpannn commented on gui/src/components/mainInput/ContinueInputBox.tsx:
```diff
@@ -112,26 +112,28 @@ interface ContinueInputBoxProps {
+      ? state.perplexityActive
+      : source === "aider"
+      ? state.aiderActive
+      : state.active;
```
please change back to switch statement

charlwillia6 replied:
The reason I had changed this is because it was adding to unnecessary re-renders when there was errors.  This was a refactor for helping with functionality, not just because I didn't like the switch statement.  This aided with fixing the error issue.

charlwillia6 replied:
Done

Fryingpannn commented:
conversation continues in: https://discord.com/channels/1237009981803462729/1296237075309269056/1304370803927683129

charlwillia6 committed:
Reverted changes to the switch statement.

charlwillia6 committed:
Removed the prop check.  No longer necessary.

charlwillia6 committed:
FIX: CMD+L/CTRL+L fixed

- Removed ignoreHighlightedCode state management because it was no longer necessary
- Reverted some refactored code to make the PR easier to read
- Removed duplicate dependencies
- Cleaned up code

charlwillia6 committed:
Improve file handling UX and fix editor persistence issues

• Add ActiveFileIndicator component
• Update TipTapEditor file handling logic
• Fix content persistence bugs
• Streamline event handling

charlwillia6 committed:
Removed spaces

charlwillia6 commented on gui/src/components/mainInput/ContinueInputBox.tsx:
```diff
@@ -156,21 +156,6 @@ const ContinueInputBox = memo(function ContinueInputBox({
     [isMainInput],
   );
 
-  // check if lastActiveIntegration === source, if so, activate gradient border and tiptap editor
```
This code is no longer necessary because it is handled in TipTapEditor

charlwillia6 committed:
Removed spaces

charlwillia6 commented on gui/src/components/mainInput/TipTapEditor.tsx:
```diff
@@ -188,15 +185,11 @@ const TipTapEditor = memo(function TipTapEditor({
     }
   );
 
-  // Create a unique key for each editor instance
```
This wasn't being used

charlwillia6 commented on gui/src/components/mainInput/TipTapEditor.tsx:
```diff
@@ -188,15 +185,11 @@ const TipTapEditor = memo(function TipTapEditor({
   const { saveSession } = useHistory(dispatch, source);
 
   const posthog = usePostHog();
-  const [isEditorFocused, setIsEditorFocused] = useState(false);
```
Wasn't being used

charlwillia6 commented on gui/src/components/mainInput/TipTapEditor.tsx:
```diff
@@ -596,38 +745,46 @@ const TipTapEditor = memo(function TipTapEditor({
-        isMetaEquivalentKeyPressed(event) &&
-        (isJetBrains() ? event.code === "KeyJ" : event.code === "KeyL")
-      ) {
-        setIgnoreHighlightedCode(true);
```
ignoreHighlightedCode and setIgnoreHighlightedCode were removed because they were no longer relevant and not needed.

charlwillia6 commented on gui/src/components/mainInput/TipTapEditor.tsx:
```diff
@@ -854,61 +1007,6 @@ const TipTapEditor = memo(function TipTapEditor({
 
   const [optionKeyHeld, setOptionKeyHeld] = useState(false);
 
-  // Use onTransaction to track content changes
```
This section of code was no longer needed as it is being handled elsewhere in TipTapEditor

charlwillia6 replied:
Look at the onUpdate handler on line 489

charlwillia6 committed:
Code clean-up for better code review

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main' into charlwillia6/add-file-to-context-when-no-text-selected

charlwillia6 committed:
Merge remote-tracking branch 'upstream/main'

charlwillia6 committed:
Merge branch 'main' into charlwillia6/add-file-to-context-when-no-text-selected
